src/FS.Physical/PhysicalFilesWatcher.cs

```diff
@@ -255,7 +255,11 @@ private void OnFileSystemEntryChange(string fullPath)
                 }
 
                 var relativePath = fullPath.Substring(_root.Length);
-                ReportChangeForMatchedEntries(relativePath);
+
+                if (relativePath != string.Empty)
+                {
```

Comment:
Wouldn't it be better to use `!string.isNullOrWhiteSpace(...)` or `!string.isNullOrEmpty(...)`?

Reply:
Could be but substring cannot return null per msdn.

Reply:
Also, iirc, a file or folder cannot terminate with a whitespace, so a whitespace only relativePath should not happen.

Reply:
I can't argue with that logic, I personally just prefer to keep it on the safe side when it comes to string empty checks.